### PR TITLE
gps-geo-action-config node extended logic and UI redesign

### PR DESCRIPTION
**Pull request by DmytroKhylko**
Changed gps-geo-action-config node logic: added toggle to switch between _On each message_ and _On first message_  , reflects back-end changes to the node ( added **reportPresenceStatusOnEachMessage** key, PR link: https://github.com/thingsboard/thingsboard/pull/9970 )
Also updated a design for the current node
Screenshot_2024-01-25_13-01

Screenshot_2024-01-25_13-03_1

Screenshot_2024-01-25_13-03

Screenshot_2024-01-25_13-04

Screenshot_2024-01-25_13-04_1

### CODE REVIEW

File: projects/rulenode-core-config/src/lib/components/action/gps-geo-action-config.component.ts [hunk 1]
```diff
@@ -28,9 +28,14 @@ export class GpsGeoActionConfigComponent extends RuleNodeConfigurationComponent
   rangeUnits = Object.keys(RangeUnit);
   rangeUnitTranslationMap = rangeUnitTranslations;
 
+  presenceMonitoringStrategies = PresenceMonitoringStrategiesData;
+  presenceMonitoringStrategyKeys = Array.from(this.presenceMonitoringStrategies.keys());
+
   timeUnits = Object.keys(TimeUnit);
   timeUnitsTranslationMap = timeUnitTranslations;
 
+  public defaultPaddingEnable = true;
```
Comment (vvlladd28) [inline projects/rulenode-core-config/src/lib/components/action/gps-geo-action-config.component.ts:37]: What is the variable of this variable if it does not change?

File: projects/rulenode-core-config/src/lib/components/action/gps-geo-action-config.component.ts [hunk 2]
```diff
@@ -62,7 +69,7 @@ export class GpsGeoActionConfigComponent extends RuleNodeConfigurationComponent
   }
 
   protected validatorTriggers(): string[] {
-    return ['fetchPerimeterInfoFromMessageMetadata', 'perimeterType'];
+    return ['fetchPerimeterInfoFromMessageMetadata', 'perimeterType', 'presenceMonitoringStrategyOnEachMessage'];
```
Comment (vvlladd28) [inline projects/rulenode-core-config/src/lib/components/action/gps-geo-action-config.component.ts]: Why add a new trigger if it doesn't change the form's validation or state?

**Review by vvlladd28 — CHANGES_REQUESTED** [review]

File: projects/rulenode-core-config/src/lib/components/action/gps-geo-action-config.component.ts (hunk 1 above)
Comment (DmytroKhylko) [reply projects/rulenode-core-config/src/lib/components/action/gps-geo-action-config.component.ts:37]: I have added the logic for this variable

File: projects/rulenode-core-config/src/lib/components/action/gps-geo-action-config.component.ts (hunk 2 above)
Comment (DmytroKhylko) [reply projects/rulenode-core-config/src/lib/components/action/gps-geo-action-config.component.ts]: I have removed 'presenceMonitoringStrategyOnEachMessage' from validatorTriggers

**Comment by ShvaykaD** [comment]
UI looks good to me. However, I think it's better to change the default config and, by default enable reportPresenceStatusOnEachMessage(currently named: presenceMonitoringStrategyOnEachMessage). This configuration is more simple and in general, I guess that it will be used more frequently than the strategy where we need to use min inside/outside durations.

**Review by vvlladd28 — APPROVED** [review]

**Pull request merged**